src/codegen/compile.ml

```diff
@@ -6000,27 +5759,8 @@ module StableMemoryInterface = struct
           [I64Type]
           Region.grow
           (fun env ->
-            (* do StableMem.grow, but detect and record change in version as well *)
-            let (set_res, get_res) = new_local64 env "size" in
             (* logical grow *)
-            StableMem.grow env ^^
-            set_res ^^
-            (* if version = version_no_stable_memory and new mem_size > 0
-               then version := version_some_stable_memory *)
-            StableMem.get_version env ^^
-            compile_eq_const StableMem.version_no_stable_memory ^^
-            StableMem.get_mem_size env ^^
-            compile_const_64 0L ^^
-            G.i (Compare (Wasm.Values.I64 I32Op.GtU)) ^^
-            G.i (Binary (Wasm.Values.I32 I32Op.And)) ^^
-            (G.if0
-               begin
-                 compile_unboxed_const StableMem.version_some_stable_memory ^^
-                 StableMem.set_version env
-               end
-               G.nop) ^^
-            (* return res *)
-            get_res))
+            StableMem.grow env))
```

Comment:
I don't quite understand why this is simpler now.
The classic compiler would just implement ExperiementalStableMemory without allocating Region metadata until forced to migrate to regions on first (userland) Region.new(). Is ESM always stored in Region0 now?

Reply:
Ok, I think I get it. Region.new in the rts looks at the virtual page size (StalbeMem.get_mem_size) directly to do the migration so don't need to track the state change here.)

Reply:
Yes, I tried to simplify this a little bit...